src/harmony-rpc.ts

```diff
@@ -21,21 +23,35 @@ export class BaseError extends Error {
 
 export class HR721 extends BaseHR721 {
   private readonly rpcProvider: JsonRpcProvider
-  constructor(provider: JsonRpcProvider) {
+  private abi: any[]
+  private readonly contract: ethers.Contract
+  constructor(provider: JsonRpcProvider, abi: any[], address: string) {
     super()
     this.rpcProvider = provider
+    this.abi = abi
+    this.contract = new ethers.Contract(address, this.abi, this.rpcProvider)
   }
 
   async balanceOf(address: string): Promise<string> {
-    if (address) {
+    if (!address) {
       throw new Error('Balance query for the zero address')
     }
```

Comment:
change the error msg to `You need to provide an address` or something like that

Reply:
Got it!